stan/math/prim/arr/err/is_matching_size.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef STAN_MATH_PRIM_ARR_ERR_IS_MATCHING_SIZE_HPP
+#define STAN_MATH_PRIM_ARR_ERR_IS_MATCHING_SIZE_HPP
+
+#include <stan/math/prim/scal/err/is_size_match.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * @return <code>true</code> if the variable sizes match
```

Comment:
Ah... this wasn't quite what I meant.

The first line should be a sentence about what it does. This line should go where it was... roughly line 23. Here, we should write:
```
* Returns <code>true</code> if the two variable sizes match.
```